processor/tailsamplingprocessor/internal/decisioncache/lru_cache.go

```diff
@@ -0,0 +1,42 @@
+package decisioncache
+
+import (
+	"encoding/binary"
+	lru "github.com/hashicorp/golang-lru/v2"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"sync"
+)
+
+// lruDecisionCache implements DecisionCache as a simple LRU cache
+type lruDecisionCache struct {
+	cache *lru.Cache[uint64, bool]
+	sync.RWMutex
+}
+
+var _ DecisionCache = (*lruDecisionCache)(nil)
+
+func NewLRUDecisionCache(size int, onEvict func(uint64, bool)) (DecisionCache, error) {
+	cache, err := lru.NewWithEvict[uint64, bool](size, onEvict)
+	if err != nil {
+		return nil, err
+	}
+	return &lruDecisionCache{cache: cache}, nil
+}
+
+func (c *lruDecisionCache) Get(id pcommon.TraceID) bool {
+	c.RLock()
+	defer c.RUnlock()
```

Comment:
Good point, I can remove the locking. This is also a great use case for benchmarking against different implementations. If the cache is locking too pessimistically we might benefit from a different library. I just used this one since it's already in the contrib project.